qiskit/algorithms/evolvers/classical_methods/scipy_imaginary_evolver.py

```diff
@@ -0,0 +1,192 @@
+    ):
+        """
+        Args:
+            timesteps: The number of timesteps to get data from the observables. At the end of each
+                       step the state will be renormalized. Note that increassing the ammount of
+                       timesteps will not increase the accuracy of the integration but can avoid
+                       overflow errors for hamiltonians with high eigenvalues.
+
+        Raises:
+            ValueError: If timesteps is not a positive integer.
+        """
+        if timesteps < 1 or not isinstance(timesteps, int):
+            raise ValueError("`timesteps` must be a positive integer.")
+
+        self.timesteps = timesteps
+
+    def evolve(self, evolution_problem: EvolutionProblem) -> EvolutionResult:
+        r"""Perform imaginary time evolution :math:`\exp(- \tau H)|\Psi\rangle`.
+
+        Evolves an initial state :math:`|\Psi\rangle` for a time :math:`\tau = it`
+        under a Hamiltonian  :math:`H`, as provided in the ``evolution_problem``.
+
+        For each timestep we use scipy's expm_multiply function to evolve the state. Internally,
+        this function will perform integration steps as well. The number of timesteps specified
+        by the user is just to renormalize the state to avoid overflow errors and to get the
+        values of the observables at those times. Note that increasing the number of timesteps
+        will not direclty increase the accuracy of the integration.
+
+        Args:
+            evolution_problem: The definition of the evolution problem.
+
+        Returns:
+            Evolution result which includes an evolved quantum state.
+
+        Raises:
+            ValueError: If a ```NaN``` is encountered in the final state of the system.
+            ValueError: If the hamiltonian in the evolution problem is time dependent.
+        """
+
+        if evolution_problem.t_param is not None:
+            raise ValueError("Time dependent hamiltonians are not currently supported.")
+
+        # Get the initial state, hamiltonian and the auxiliary operators.
+        state, hamiltonian, aux_ops, aux_ops_2, timestep = self._start(
+            evolution_problem=evolution_problem
+        )
+
+        # Create empty arrays to store the time evolution of the aux operators.
+        operator_number = (
+            0 if evolution_problem.aux_operators is None else len(evolution_problem.aux_operators)
+        )
+        ops_ev_mean = np.empty(shape=(operator_number, self.timesteps + 1), dtype=float)
+
+        ops_ev_std = np.empty(shape=(operator_number, self.timesteps + 1), dtype=float)
+
+        for ts in range(self.timesteps):
+            (ops_ev_mean[:, ts], ops_ev_std[:, ts],) = self._evaluate_aux_ops(
+                aux_ops,
+                aux_ops_2,
+                state,
+            )
+            state = expm_multiply(A=-hamiltonian * timestep, B=state)
+            if np.nan in state:
+                raise ValueError(
```

Comment:
Would a `RuntimeError` be better here since this happens during algorithm execution?
```suggestion
                raise RuntimeError(
```

Reply:
Thank you!